tests_cpp/eigen_2d_euler_parameters/main.cc

```diff
@@ -0,0 +1,190 @@
+
+    for (int api=0; api<4; ++api){
+      if      (api==0){ std::cout << "\nNormalShock: testing overload: default params \n"; }
+      else if (api==1){ std::cout << "\nNormalShock: testing overload: default params, customBCs \n"; }
+      else if (api==2){ std::cout << "\nNormalShock: testing overload: default params, icFlag = 1 \n"; }
+      else if (api==3){ std::cout << "\nNormalShock: testing overload: default params, icFlag = 1, customBCs \n"; }
+
+      std::unordered_map<std::string, double> gold;
+      gold["gamma"] = 1.4;
+      gold["mach"] = 9.;
+
+      if (api==0){
+	auto appObj = pda::create_problem_eigen(meshObj, probId, flsc);
+	if (!check(appObj, gold)){ std::puts("FAILED"); return 0; }
+      }
+
+      else if (api==1){
+	auto noop = pda::impl::NoOperation<void>();
+	auto appObj = pda::create_problem_eigen(meshObj, probId, flsc,
+						noop, noop, noop, noop);
+	if (!check(appObj, gold)){ std::puts("FAILED"); return 0; }
+      }
+
+      else if (api==2){
+	auto appObj = pda::create_problem_eigen(meshObj, probId, flsc, 1);
+	if (!check(appObj, gold)){ std::puts("FAILED"); return 0; }
+      }
+
+      else if (api==3){
+	auto noop = pda::impl::NoOperation<void>();
+	auto appObj = pda::create_problem_eigen(meshObj, probId, flsc,
+						noop, noop, noop, noop, 1);
+	if (!check(appObj, gold)){ std::puts("FAILED"); return 0; }
+      }
+    }
+  }
+
+  {
+    const auto probId = pda::Euler2d::NormalShock;
+    const int icFlag = 1;
+    for (int api=0; api<4; ++api){
+      if      (api==0){ std::cout << "\nNormalShock: testing overload: custom params, icFlag =1, customBCs \n"; }
+
+      // loop over each param and only change one at a time
+      for (int i=0; i<2; ++i)
+      {
+	std::unordered_map<std::string, double> myparams;
+	if (i==0) myparams["gamma"] = 1.8;
+	if (i==1) myparams["mach"]  = 12.;
+
+	std::unordered_map<std::string, double> gold;
+	gold["gamma"]  = (i==0) ? 1.8 : 1.4;
+	gold["mach"] = (i==1) ? 12. : 9.0;
+
+	if (api==0){
+	  auto noop = pda::impl::NoOperation<void>();
+	  auto appObj = pda::create_problem_eigen(meshObj, probId, flsc,
+						  noop, noop, noop, noop, icFlag, myparams);
+	  if (!check(appObj, gold)){ std::puts("FAILED"); return 0; }
+	}
+      }
+    }
+  }
+
+  {
+    const auto probId = pda::Euler2d::Riemann;
+
+    for (int api=0; api<6; ++api){
+      if      (api==0){ std::cout << "\nRiemann: testing overload: default params \n"; }
+      else if (api==1){ std::cout << "\nRiemann: testing overload: default params, customBCs \n"; }
+      else if (api==2){ std::cout << "\nRiemann: testing overload: default params, icFlag = 1 \n"; }
+      else if (api==3){ std::cout << "\nRiemann: testing overload: default params, icFlag = 1, customBCs \n"; }
+      else if (api==4){ std::cout << "\nRiemann: testing overload: default params, icFlag = 2 \n"; }
+      else if (api==5){ std::cout << "\nRiemann: testing overload: default params, icFlag = 2, customBCs \n"; }
```

Comment:
you forgot to update the test to make sure it works correctly for the new parameters you added

Reply:
Thanks, updated in [177495b](https://github.com/Pressio/pressio-demoapps/pull/157/commits/177495b7af7617e6e2952f275b09532fe3cc3163)